Freemium: update marketplace product card label

raicem opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Change the label for Freemium products in the In-App Marketplace.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

1. Checkout this branch
2. In the `product-card-footer.tsx`, set `product.freemium_type = 'primary';` [in line 19. ](https://github.com/woocommerce/woocommerce/blob/71ebf5932437becb72e4fb65e2014aaf038d5bf9/plugins/woocommerce-admin/client/marketplace/components/product-card/product-card-footer.tsx#L18)
     - This eliminates the need for connecting to a local WooCommerce.com database. The product endpoint doesn't yet have any freemium products. 
3. Get a new build
4. Go to the **WooCommerce > Extensions > Browse** and see the new "Free plan available" label (you may need to scroll down a bit as most products have **Add to Store** button still)
    ![image](https://github.com/woocommerce/woocommerce/assets/4509348/0b1a3630-8779-419f-aa1b-21a7a08080d9)
5. Please test around this issue

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message Show the label for freemium products in the in-app marketpalce

#### Comment 

</details>

raicem committed:
Freemium: update marketplace product card label

kdevnel commented on plugins/woocommerce-admin/client/marketplace/components/product-card/product-card-footer.tsx:
```diff
@@ -83,17 +107,10 @@ function ProductCardFooter( props: { product: Product } ) {
-					{ product.price === 0
-						? ''
-						: __( ' annually', 'woocommerce' ) }
+					{ getBillingText() }
```
This is so much better!

kdevnel reviewed (approved):
Tests well and works as expected. Thank you!

Just note, there are no links to the changes you mention in step 4 for the cache values. I didn't need them, but thought to mention it. Also, I only realised after reading the code that testing with a free product set as primary wouldn't work. 

PS - you have some E2E test failures, might be worth running them again.

raicem commented:
> Just note, there are no links to the changes you mention in step 4 for the cache values. I didn't need them, but thought to mention it.

Thank you @kdevnel, I have fixed in case it's needed in the future. 

I have rebased this on top of trunk and hopefully that'll resolve the e2e tests :crossed_fingers:

kdevnel reviewed (approved):
Looks good! Now it seems the auto-changelog adder is failing to run. You could always add the changelog manually.

raicem commented:
> Looks good! Now it seems the auto-changelog adder is failing to run. You could always add the changelog manually.

Yeah, I'm struggling with that now. I don't want to add a changelog item, I think this change is not worth it. I wanted to do a patch buy I haven't picked the **Type**. Maybe that's why. Re-running it now.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

Commit:
Add changefile(s) from automation for the following project(s): woocommerce
